apps/greenhouse-management/src/components/Auth.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useState } from "react"
+import { Spinner, Stack } from "juno-ui-components"
+import { useAuthAppLoaded, useIsLoggedIn } from "./StoreProvider"
+import { Transition } from "@tailwindui/react"
+
+// Adds a loading screen while during auth
+// Shows children when auth is complete
+
+const Auth = ({ children }) => {
+  const authAppLoaded = useAuthAppLoaded()
+  const authLoggedIn = useIsLoggedIn()
+
+  const [loading, setLoading] = useState(!authAppLoaded)
+
+  // timeout for waiting for auth
+  useEffect(() => {
+    setLoading(!authAppLoaded)
+    if (authAppLoaded) return
+    // set timeout for waiting for auth app
+    let loadingTimer
+    if (!authAppLoaded) {
+      loadingTimer = setTimeout(() => {
+        if (!authAppLoaded) setLoading(false)
+      }, 30000) // 30 seconds
+    }
+
+    return () => loadingTimer && clearTimeout(loadingTimer)
+  }, [authAppLoaded, setLoading])
+
+  return (
+    <>
+      <Transition
```

Comment:
We don't need any transtition. This is just something Andreas tested out in greenhouse. As far as I remember we decide to avoid transitions when posible.